### PR TITLE
Query appointment data 

### PR DESCRIPTION
**Pull request by PormpimonPang**
Data that used in appointment page
• Date
• Time
• Reason
• Preparation

### CODE REVIEW

File: lib/widget/show_appointment.dart [hunk 1]
```diff
@@ -29,59 +29,114 @@ class ShowAppointmentState extends State<ShowAppointment> {
...
-                          Text(
-                            'นัดมาตรวจ: ${appointment['reason']}',
-                            style: Theme.of(context).textTheme.bodyText1,
-                          ),
-                          Text(
-                            'การเตรียมความพร้อม: ${appointment['preparation']}',
-                            style: Theme.of(context).textTheme.bodyText1,
-                          ),
+                          CircularProgressIndicator(),
+                          Text('Loading...'),
```
Comment (Thaweesak-sai) [inline lib/widget/show_appointment.dart]: Why do we need these 2 Builder?

File: lib/widget/show_appointment.dart [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                  if (!anSubCollection.hasData) {
+                    return Center(
+                      child: Column(
+                        children: <Widget>[
+                          CircularProgressIndicator(),
+                          Text('Loading...'),
+                        ],
+                      ),
+                    );
+                  }
```
Comment (Thaweesak-sai) [inline lib/widget/show_appointment.dart:74]: Why do we need these 2 builders

File: lib/widget/show_appointment.dart [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                                  padding: const EdgeInsets.only(left: 20),
+                                  child: Column(
+                                    crossAxisAlignment:
+                                        CrossAxisAlignment.stretch,
+                                    children: <Widget>[
+                                      Text(
+                                        'รายละเอียดการนัดหมาย',
+                                        style: Theme.of(context)
+                                            .textTheme
+                                            .bodyText1,
```
Comment (Thaweesak-sai) [inline lib/widget/show_appointment.dart:99]: So, the UI gonna be changed ?

File: lib/widget/show_appointment.dart (hunk 2 above)
Comment (PormpimonPang) [reply lib/widget/show_appointment.dart:74]: The builder will load on that page, equivalent to hot reload.

File: lib/widget/show_appointment.dart (hunk 3 above)
Comment (PormpimonPang) [reply lib/widget/show_appointment.dart:99]: yes

File: lib/widget/show_appointment.dart (hunk 2 above)
Comment (Thaweesak-sai) [reply lib/widget/show_appointment.dart:74]: But we don't need other data except the data from `Appointments` Collection.

Comment (Thaweesak-sai) [reply lib/widget/show_appointment.dart:74]: In line 63, it's gonna consume the query resource

File: lib/widget/show_appointment.dart (hunk 3 above)
Comment (PormpimonPang) [reply lib/widget/show_appointment.dart:99]: Screen Shot 2564-02-03 at 10 02 06

appointment will change without hot reload

File: lib/widget/show_appointment.dart (hunk 2 above)
Comment (PormpimonPang) [reply lib/widget/show_appointment.dart:74]: If there is no builder for loading, it will need to be hot reload again.

Comment (PormpimonPang) [reply lib/widget/show_appointment.dart:74]: I'll try to fix it again.

**Review by Thaweesak-sai — APPROVED** [review]

**Pull request merged**